timeboost/src/conf.rs

```diff
@@ -64,8 +67,95 @@ pub struct TimeboostConfig {
 }
 
 impl TimeboostConfig {
-    pub fn sequencer_config(&self) -> SequencerConfig {
-        SequencerConfig::builder()
+    pub async fn sequencer_config(&self) -> Result<SequencerConfig> {
```

Comment:
Please don't. This config is meant to be a simple data-only structure, not involving I/O. I also can't help but notice that the contract types are leaking everywhere. There is a `ProviderBuilder` in here, in the timeboost binary and in the timeboost lib.rs. Why can this not be encapsulated in one module? Timeboost only wants two things (a) the current committee and (b) an eventual notification of the next committee. Having all the contract implementation details everywhere should not be necessary IMHO.